frontend/src/components/resourceMap/GraphView.tsx

```diff
@@ -0,0 +1,390 @@
+                    sx={{
+                      lineHeight: '1',
+                    }}
+                  />
+                )}
+
+                <Chip
+                  label={t('Group By: {{ name }}', { name: t('Instance') })}
+                  color={groupBy === 'instance' ? 'primary' : undefined}
+                  variant={groupBy === 'instance' ? 'filled' : 'outlined'}
+                  icon={groupBy === 'instance' ? <Icon icon="mdi:check" /> : undefined}
+                  onClick={() => setGroupBy(it => (it === 'instance' ? undefined : 'instance'))}
+                  sx={{
+                    lineHeight: '1',
+                  }}
+                />
+
+                <Chip
+                  label={t('Group By: {{ name }}', { name: t('Node') })}
+                  color={groupBy === 'node' ? 'primary' : undefined}
+                  variant={groupBy === 'node' ? 'filled' : 'outlined'}
+                  icon={groupBy === 'node' ? <Icon icon="mdi:check" /> : undefined}
+                  onClick={() => setGroupBy(it => (it === 'node' ? undefined : 'node'))}
+                  sx={{
+                    lineHeight: '1',
+                  }}
+                />
+
+                <Chip
+                  label={t('Status: Error or Warning')}
+                  color={hasErrorsFilter ? 'primary' : undefined}
+                  variant={hasErrorsFilter ? 'filled' : 'outlined'}
+                  icon={hasErrorsFilter ? <Icon icon="mdi:check" /> : undefined}
+                  onClick={() => setHasErrorsFilter(!hasErrorsFilter)}
+                  sx={{
+                    lineHeight: '1',
+                  }}
+                />
+
+                {graphSize < 50 && (
+                  <Chip
+                    label={t('Expand All')}
+                    color={expandAll ? 'primary' : undefined}
+                    variant={expandAll ? 'filled' : 'outlined'}
+                    icon={expandAll ? <Icon icon="mdi:check" /> : undefined}
+                    onClick={() => setExpandAll(it => !it)}
+                    sx={{
+                      lineHeight: '1',
+                    }}
+                  />
+                )}
+              </Box>
+            </Box>
+            <div style={{ flexGrow: 1 }}>
+              <GraphRenderer
+                nodes={layoutedGraph.nodes}
+                edges={layoutedGraph.edges}
+                onMoveStart={e => {
+                  if (e === null) return;
+                  viewportMovedRef.current = true;
+                }}
+                onBackgroundClick={() => {
+                  if (selectedGroup && selectedNodeId === selectedGroup.id) {
+                    const parent = getParentNode(fullGraph, selectedGroup.id);
+                    if (parent) {
+                      setSelectedNodeId(parent.id);
+                    }
+                  } else if (selectedGroup) {
+                    setSelectedNodeId(selectedGroup.id);
+                  }
```

Comment:
How about?

```typescript
if (selectedGroup) {
  let nodeId = selectedGroup.id;
  if (selectedNodeId === nodeId) {
    const parent = getParentNode(fullGraph, selectedGroup.id);
    if (parent) {
      nodeId = parent.id;
    }
  }
  setSelectedNodeId(nodeId);
}
```

Reply:
that whole logic block was not necessary, replaced it with something simpler

Reply:
Not sure where this is... @joaquimrocha can you please confirm the change?

Reply:
It wasn't moved, it was just replaced with a couple of lines in the same place. GraphView.tsx:283

Reply:
@joaquimrocha and I had a call. Changed the logic here to remove "go back when on background click" behavior, now it will only close the side panel on background click